apps/extension/src/routes/page/onboarding/generate.tsx

```diff
@@ -14,20 +14,74 @@ import { generateSelector } from '../../../state/seed-phrase/generate';
 import { usePageNav } from '../../../utils/navigate';
 import { PagePath } from '../paths';
 import { WordLengthToogles } from '../../../shared/containers/word-length-toogles';
+import { ChainRegistryClient } from '@penumbra-labs/registry';
+import { fetchBlockHeight } from '../../../hooks/full-sync-height';
+import { localExtStorage } from '../../../storage/local';
+import { sample } from 'lodash';
+
+const fetchBlockHeightWithFallback = async (endpoints: string[]): Promise<number> => {
+  if (endpoints.length === 0) {
+    throw new Error('All RPC endpoints failed to fetch the block height.');
+  }
+
+  // Randomly select an rpc endpoint from the chain registry
+  const randomEndpoint = sample(endpoints);
+
+  try {
+    const walletCreationBlockHeight = await fetchBlockHeight(randomEndpoint!);
+    if (walletCreationBlockHeight !== undefined) {
+      return walletCreationBlockHeight;
+    } else {
+      // Remove the current endpoint from the list and try again
+      const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomEndpoint);
+      return fetchBlockHeightWithFallback(remainingEndpoints);
+    }
+  } catch (error) {
+    const remainingEndpoints = endpoints.filter(endpoint => endpoint !== randomEndpoint);
+    return fetchBlockHeightWithFallback(remainingEndpoints);
+  }
+};
```

Comment:
this fetches the block height by randomly querying one of the rpc endpoints in the chain registry. something to consider is whether this operation is possibly blocking or if the rpc call will eventually time out?

Reply:
I think we added timeouts to our promise clients (?). It's async, so not thread blocking, but we should have a think whether this should block the onboarding step.

Reply:
as you pointed out, while this approach is non-blocking at the thread level,  it could still delay onboarding if the rpc calls take too long. Consequently, I think we should implement a timeout and a retry limit.

Reply:
cc @turbocrime